src/SurveyService.Data/QuestionRepository.cs

```diff
@@ -11,6 +13,32 @@ public class QuestionRepository : IQuestionRepository
 {
   private readonly IDataProvider _provider;
 
+  private IQueryable<DbQuestion> CreateGetPredicates(
+    GetQuestionFilter filter,
+    IQueryable<DbQuestion> dbQuestions) 
+  {
+    if (filter.IncludeOptions)
+    {
+      if (filter.IncludeAnswers)
+      {
+        dbQuestions = dbQuestions?
+          .Include(question => question.Options.Where(option => option.IsActive))
+          .ThenInclude(option => option.UsersAnswers);
+      }
+      else
+      {
+        dbQuestions = dbQuestions?
+          .Include(question => question.Options.Where(option => option.IsActive && !option.IsCustom));
+      }
+    }
+    else
+    {
+      dbQuestions = dbQuestions?.Where(question => question.Id == filter.QuestionId);
```

Comment:
what is it for here? you've already filtered by QuestionId in Get

Reply:
Done